src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -207,4 +178,25 @@ def self.client_get(request_options)
   rescue Java::ComPuppetlabsHttpClient::HttpClientException => e
     raise Puppet::Server::HttpClientError.new(e.message, e)
   end
+
+
+  def create_session
+    raise NotImplementedError
+  end
+
+  def connect(uri, options: {}, &block)
+    raise NotImplementedError
+  end
+
+  def head(url, headers: {}, params: {}, options: {})
+    raise NotImplementedError
+  end
+
+  def put(url, headers: {}, params: {}, options: {})
+    raise NotImplementedError
+  end
+
+  def delete(url, headers: {}, params: {}, options: {})
+    raise NotImplementedError
+  end
```

Comment:
Why are all these needed?

Reply:
I ended up making this class inherit from `Puppet::HTTP::Client` in order to get access to its [URL encoding functions](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/http/client.rb#L279-L285). That class has implementations for these functions that aren't valid inside Puppet Server. 

I'm open to alternatives if people don't like that solution, though I don't really want to duplicate the code. But maybe there's a refactor in Puppet that we want to do to make those more shareable?

Reply:
ahh, I didn't realize in my re-review that this wasn't the response class that was already inheriting, gotcha. it's really easy to skim PRs I've looked at already.. 🙃

Reply:
Sorry, I should have made a separate commit!

Reply:
This is good because it makes it explicit that those methods aren't supported. We were just getting lucky before, because if anyone did call `head`, etc then they'd get NoMethodError... but it was hard to tell just by looking at the source.